### ISSUE TITLE
require-atomic-updates misses a captured shared property written back after await

### ISSUE THREAD

**Issue by fireairforce** [utooland/utoo-lint#1229]
## Summary

Reproduced with @utoo/lint 0.4.1 on Node.js 22.23.1, macOS arm64. Concurrent calls can read the same count before awaiting and overwrite each other with previous + 1 afterwards. The shared-property race should be reported.

## Minimal reproduction

```sh
npm init -y
npm install --save-dev @utoo/lint@0.4.1 eslint@8.57.1 @typescript-eslint/parser@6.21.0 
```

`example.tsx`:

```tsx
const state = { count: 0 };
export async function update(task) {
  const previous = state.count;
  await task();
  state.count = previous + 1;
}
```

`utlint.config.json`:

```json
{
  "rules": {
    "require-atomic-updates": "error"
  }
}
```

`.eslintrc.cjs`:

```js
module.exports = {
  "root": true,
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "ecmaVersion": 2022,
    "sourceType": "module",
    "ecmaFeatures": {
      "jsx": true
    }
  },
  "rules": {
    "require-atomic-updates": "error"
  }
};
```

```sh
./node_modules/.bin/utlint --config utlint.config.json --format=json example.tsx
./node_modules/.bin/eslint --no-eslintrc --config .eslintrc.cjs --format=json example.tsx
```

## Result

ESLint reports one require-atomic-updates error on state.count = previous + 1. utlint reports no diagnostics.

Both tools check the same file successfully; there are no syntax errors or lint-disable comments.

Control cases: both tools report `state.count += await read()`, and both accept a write after `await` that does not depend on an earlier read. The missing case is the read captured before `await` and used afterwards.
